pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -52,18 +52,21 @@ func (os *OpenStack) AttachDisk(instanceID string, diskName string) (string, err
 		if instanceID == disk.Attachments[0]["server_id"] {
 			glog.V(4).Infof("Disk: %q is already attached to compute: %q", diskName, instanceID)
 			return disk.ID, nil
-		} else {
-			errMsg := fmt.Sprintf("Disk %q is attached to a different compute: %q, should be detached before proceeding", diskName, disk.Attachments[0]["server_id"])
-			glog.Errorf(errMsg)
-			return "", errors.New(errMsg)
+		}
+
+		glog.V(2).Infof("Disk %q is attached to a different compute (%q), detaching", diskName, disk.Attachments[0]["server_id"])
+		err = os.DetachDisk(fmt.Sprintf("%s", disk.Attachments[0]["server_id"]), diskName)
```

Comment:
I'm not super familiar with the release process here but 100% agreed with @codablock that this should not be pulled into mainline kube in its current state. For an example, lets say you're doing a `kubectl delete` on a webserver that's mid transaction. My expected behavior would be for the volume to be available until my pod fully exits, so the server can complete its final set of transactions. With this patch it would lose access to its volumes the instant a new pod gets scheduled which could cause data loss.

I'd imagine there should be a check to the kube API before this detach call to verify that the volume is not in use by any active pods. If this check was added kube would simply keep retrying until the volume isn't in use, at which point it would detach and continue.